webapp/src/App.tsx

```diff
@@ -1,22 +1,42 @@
 import React, {useEffect} from 'react';
+import {useDispatch} from 'react-redux';
 
-import Constants from 'constants/index';
 import usePluginApi from 'hooks/usePluginApi';
 
 //global styles
 import 'styles/main.scss';
+import {pluginApiServiceConfigs} from 'constants/apiService.constant';
+import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
+
+import {setConnected} from 'reducers/connectedState';
+import {defaultPage, defaultPerPage} from 'constants/common.constants';
 
 /**
  * This is main App component for plugin
  * @returns {JSX.Element}
  */
 const App = (): JSX.Element => {
-    const {makeApiRequestWithCompletionStatus} = usePluginApi();
+    const dispatch = useDispatch();
+    const {makeApiRequestWithCompletionStatus, getApiState} = usePluginApi();
 
     useEffect(() => {
-        makeApiRequestWithCompletionStatus(Constants.pluginApiServiceConfigs.whitelistUser.apiServiceName);
+        const linkedChannelsParams: SearchLinkedChannelParams = {page: defaultPage, per_page: defaultPerPage};
+
+        makeApiRequestWithCompletionStatus(pluginApiServiceConfigs.whitelistUser.apiServiceName);
+        makeApiRequestWithCompletionStatus(pluginApiServiceConfigs.needsConnect.apiServiceName);
+        makeApiRequestWithCompletionStatus(pluginApiServiceConfigs.getLinkedChannels.apiServiceName, linkedChannelsParams);
```

Comment:
why are we making an API call here irrespective of user connection?

Reply:
Discussed